pre_commit_hooks/check_nginx_wide_range.py

```diff
@@ -0,0 +1,256 @@
+
+    Args:
+      config: dict with current nginx config content
+      name: searched directive name
+      value: if passed search for concrete directive `name` with exact `value`
+      results: list with `search` results
+
+    Returns:
+      (list): list with found directives configs
+
+    """
+    results = results or []
+    if config["directive"] == name:
+        if value is None or config["args"] == value:
+            results.append(config)
+
+    blocks = config.get("block", [])
+    for block in blocks:
+        block["file"] = config.get("file")
+        results = _search_directive(block, name, value, results)
+    return results
+
+
+def _disabled_locations_exist(locations: List[Dict] | None = None) -> bool:
+    """Check that all corresponding `disabled` locations were added.
+
+    Search for below directives in nginx config, if at least one of them
+    is not defined - error will be raised:
+
+        location ~ \.(json|sh|conf|xml|md|conf|toml|yml|yaml|log|pid)$  {deny all;}
+        location ~ /cron.*                                              {deny all;}
+        location ~ /\.                                                  {deny all;}
+        location ~ autodiscover.xml {return 403;}
+        location ~ apple-touch-icon {return 403;}
+        location ~ ^/(app/|vendor|src|tests|vagrant|docs|phpunit|svn|git|docker|migrations|Makefile) {return 403;}
+
+    Args:
+      locations: list with all `locations` directives configs found in nginx config
+
+    Returns:
+      (bool): flag whether all required disabled `locations` directives are added
+
+    """
+    locations = locations or []
+
+    # locations args that should exactly match
+    exact_match = [
+        dict(regex=r"^{item}$", values=[item[0]], expected=item[1], exists=False)
+        for item in [
+            ("/cron\.\*", "/cron.*"),
+            ("/\\\.", "/\."),
+            ("autodiscover\.xml", "autodiscover.xml"),
+            ("apple-touch-icon", "apple-touch-icon"),
+        ]
+    ]
+
+    # location args that may be not exactly matched, checked by regex
+    regex_match = [
+        dict(
+            regex=r"^\\\.\(.*{item}.*\)\$$",
+            values=[
+                "json",
+                "sh",
+                "xml",
+                "md",
+                "conf",
+                "toml",
+                "yml",
+                "yaml",
+                "log",
+                "pid",
+            ],
+            expected="\.(json|sh|xml|md|conf|toml|yml|yaml|log|pid)$",
+            exists=False,
+        ),
+        dict(
+            regex=r"^\^/\(.*{item}.*\)",
+            values=[
+                "app/",
+                "vendor",
+                "src",
+                "tests",
+                "vagrant",
+                "docs",
+                "phpunit",
+                "svn",
+                "git",
+                "docker",
+                "migrations",
+                "Makefile",
```

Comment:
I think to make it better - we should pass it as an argument into the hook. And if not passed - use default. Then we can control what to verify

Reply:
done